app/queries/district_analytics_query.rb

```diff
@@ -25,52 +26,40 @@ def call
   def results
     results = [
       registered_patients_by_period,
-      total_patients,
-      patients_with_bp_by_period
+      total_registered_patients,
+      follow_up_patients_by_period
     ].compact
 
     return {} if results.blank?
     results.inject(&:deep_merge)
   end
 
-  def total_patients
+  def total_registered_patients
     @total_patients ||=
       Patient
         .with_hypertension
-        .joins(:assigned_facility)
+        .joins(:registration_facility)
         .where(facilities: {id: facilities})
         .group("facilities.id")
         .count
 
     return if @total_patients.blank?
 
     @total_patients
-      .map { |facility_id, count| [facility_id, {total_patients: count}] }
+      .map { |facility_id, count| [facility_id, {total_registered_patients: count}] }
       .to_h
   end
 
   def registered_patients_by_period
-    @registered_patients_by_period ||=
-      Patient
-        .with_hypertension
-        .joins(:registration_facility)
-        .where(facilities: {id: facilities})
-        .group("facilities.id")
-        .group_by_period(@period, :recorded_at)
-        .count
+    result = ActivityService.new(region, last: nil, group: [:registration_facility_id]).registrations
 
-    group_by_facility_and_date(@registered_patients_by_period, :registered_patients_by_period)
+    group_by_date_and_facility(result, :registered_patients_by_period)
   end
 
-  def patients_with_bp_by_period
-    @patients_with_bp_by_period ||=
-      Patient
-        .group("assigned_facility_id")
-        .where(assigned_facility: facilities)
-        .hypertension_follow_ups_by_period(@period, last: @prev_periods)
-        .count
+  def follow_up_patients_by_period
+    result = ActivityService.new(region, last: nil, group: BloodPressure.arel_table[:facility_id]).follow_ups
 
-    group_by_facility_and_date(@patients_with_bp_by_period, :patients_with_bp_by_period)
+    group_by_date_and_facility(result, :follow_up_patients_by_period)
```

Comment:
Instead of all this in-memory aggregation, do you think we could just run another `ActivityService` query? Perhaps...

```ruby
ActivityService.new(region, last: nil).follow_ups
```

...perhaps no `group` will return results that are grouped only by date, which is what we want here?